vm/src/obj/objdict.rs

```diff
@@ -284,6 +247,104 @@ impl ItemProtocol for PyDictRef {
     }
 }
 
+macro_rules! dict_iterator {
```

Comment:
Not sure whether I like this amount of macro-fu, but still, it solves the duplication. Probably this thing has too many macro arguments. Another idea, create a struct which takes a trait which handles the specifics?

Reply:
I'm not sure I see how to get much of the benefit here with traits, especially while avoiding wrong abstraction. The macro feels like the most straight-forward approach.